src/Arcus.Security.Secrets.Core/Exceptions/SecretNotFoundException.cs

```diff
@@ -32,9 +40,32 @@ public SecretNotFoundException(string name, Exception innerException) : base($"T
             Name = name;
         }
 
+        /// <summary>
+        /// Creates <see cref="SecretNotFoundException"/> used for serialization.
+        /// </summary>
+        /// <param name="info">The <see cref="T:SerializationInfo"></see> that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">The <see cref="T:StreamingContext"></see> that contains contextual information about the source or destination.</param>
+        protected SecretNotFoundException(SerializationInfo info, StreamingContext context) : base(info, context)
+        {
+            Name = info.GetString(nameof(Name));
+        }
+
         /// <summary>
         /// Name of the missing key
         /// </summary>
         public string Name { get; }
```

Comment:
Let's add `Undefined` or something else that is explicit that it wasn't specified

Reply:
Not sure what you mean? Updating the exception message?

Reply:
No, assign a default value to `Name` with `Undefined` in case people call the default constructor

Reply:
Aha, ok. Yes!